File: ruby/testcode/ssti_036.rb
require_relative 'shared'

begin
  require 'slim'
rescue LoadError
end

# vuln-code-snippet start ruby_ssti_slim_fixed_file
def handler(req)
  output = Slim::Template.new('views/page.slim').render(Object.new) # vuln-code-snippet safe-line ruby_ssti_slim_fixed_file
  BenchmarkResponse.html(output)
end
# vuln-code-snippet end ruby_ssti_slim_fixed_file

File: php/testcode/ssti_036.php
<?php
require_once __DIR__ . '/shared.php';

// vuln-code-snippet start php_ssti_plates_static_dir
function ssti036(BenchmarkRequest $req): BenchmarkResponse {
    $templates = new League\Plates\Engine(__DIR__ . '/views');
    $html = $templates->render('profile', ['user' => $req->param('name')]); // vuln-code-snippet safe-line php_ssti_plates_static_dir
    return BenchmarkResponse::ok($html);
}
// vuln-code-snippet end php_ssti_plates_static_dir
